### PR TITLE
Support more thorough device identification

### PR DESCRIPTION
**Pull request by tstellanova**

### CODE REVIEW

**Comment by jonas-schievink** [comment]
Thanks! Doesn't the meaning of the company ID code depend on the number of continuation bytes before it though?

**Comment by tstellanova** [comment]
> Thanks! Doesn't the meaning of the company ID code depend on the number of continuation bytes before it though?

What we could do is handle this similar to the description here:
http://infocenter.arm.com/help/index.jsp?topic=/com.arm.doc.faqs/ka14408.html
as in, store a count of continuation codes (which indicates which bank the manufacturer's code is in). This could be a separate field in the `Identification` struct, and consumers of the API could check it if desired.

**Comment by jonas-schievink** [comment]
Yeah that sounds like a good idea then! I was thinking of removing this feature altogether (https://github.com/jonas-schievink/spi-memory/issues/6), but it seems like it has some uses, so if we can fix it and make it actually useful then even better.

File: src/series25.rs [hunk 1]
```diff
@@ -15,13 +15,24 @@ pub struct Identification {
     /// device ID. How many bytes are used depends on manufacturer's place in
     /// the JEDEC list, I guess.
```
Comment (jonas-schievink) [inline src/series25.rs]: This comment is now outdated

**Review by jonas-schievink — COMMENTED** [review]
Please put a space after the `//` in comments. Other than that (and the outdated comment) this looks good now, thanks!

File: src/series25.rs [hunk 2]
```diff
@@ -222,3 +266,24 @@ impl<SPI: Transfer<u8>, CS: OutputPin> BlockDevice<u32, SPI, CS> for Flash<SPI,
         Ok(())
     }
 }
+
+// TODO reintroduce test
+// Manually ran this test by tweaking [dev-dependencies] and config;
+// however, until we can specify separate dependencies for
+// tests and examples, this will not be runnable in automation.
```
Comment (jonas-schievink) [inline src/series25.rs]: This looks good, what's the issue with the test? Normally `no_std` libraries can be made testable by replacing the `#![no_std]` with `#![cfg_attr(not(test), no_std)]`.

Comment (tstellanova) [reply src/series25.rs]: The issue is apparently that `[dev-dependencies]` applies to both test and the `dump` example, which pulls in eg `stm32f4xx-hal`, and there's no way to define separate dependencies for test.  `series25.rs` doesn't include `#![no_std]` currently.  If you simply include this one test and run:
`cargo test --package spi-memory --lib series25::tests`

You get the error:
```
error[E0463]: can't find crate for `test`
   --> src/series25.rs:280:5
    |
280 | /     fn test_decode_jedec_id() {
281 | |         let cypress_id_bytes = [0x7F, 0x7F, 0x7F, 0x7F, 0x7F, 0x7F, 0xC2, 0x22, 0x08];
282 | |         let ident = Identification::from_jedec_id(&cypress_id_bytes);
283 | |         assert_eq!(0xC2, ident.mfr_code());
...   |
287 | |         assert_eq!(device_id[1], 0x08);
288 | |     }
    | |_____^ can't find crate
```

**Comment by jonas-schievink** [comment]
For some reason the CI status doesn't show up here, but it's passing: https://travis-ci.org/github/jonas-schievink/spi-memory/builds/666952763

Thanks for the PR!

**Pull request merged**